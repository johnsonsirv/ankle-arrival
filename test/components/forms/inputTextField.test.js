/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import InpuTextField from '../../../src/components/forms/inputs/inputTextField';

function setup() {
  const props = {
    name: 'description',
    onChange: jest.fn(),
  };
  const enzymeWrapper = mount(<InpuTextField {...props} />);

  return { props, enzymeWrapper };
}

describe('<InpuTextField /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('renders correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render 1 editable <input> text', () => {
    expect(enzymeWrapper.find('input').prop('readOnly')).toBeUndefined();
  });
});

describe('<InputTextFieldd /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('should handle onChange', () => {
    enzymeWrapper.simulate('change', { target: { value: 'Sample Text' } });
    expect(props.onChange).toHaveBeenCalled();
  });
});
