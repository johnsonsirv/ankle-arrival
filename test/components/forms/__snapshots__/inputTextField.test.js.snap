// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InpuTextField /> rendering renders correctly 1`] = `ReactWrapper {}`;
